File: AppOptimize/app/src/main/java/com/gl/appoptimize/OptimizeActivity.java
package com.gl.appoptimize;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import androidx.appcompat.app.AppCompatActivity;

public class OptimizeActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_optimize);

    }
    public void toLeakActivity(View view) {
        startActivity(new Intent(this, LeakActivity.class));
    }

    public void toBlockActivity(View view) {
        startActivity(new Intent(this, BlockActivity.class));
    }


}

File: AppOptimize/app/src/main/java/com/gl/appoptimize/OptimizeActivity.kt
package com.gl.appoptimize

import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.gl.appoptimize.block.BlockActivity
import com.gl.appoptimize.memory.LeakActivity

class OptimizeActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_optimize)
    }

    @Synchronized
    fun toLeakActivity(view: View?) {
        startActivity(Intent(this, LeakActivity::class.java))
    }

    fun toBlockActivity(view: View?) {
        synchronized(this) { startActivity(Intent(this, BlockActivity::class.java)) }
    }

    fun toCrashActivity(view: View?) {
        synchronized(this) { startActivity(Intent(this, BlockActivity::class.java)) }
    }
}